src/Lexer.php

```diff
@@ -175,8 +255,17 @@ protected function lexData()
 
         // push the template text first
         $text = $textContent = substr($this->code, $this->cursor, $position[1] - $this->cursor);
+
+        // trim?
         if (isset($this->positions[2][$this->position][0])) {
-            $text = rtrim($text);
+            if ($this->options['whitespace_trim'] === $this->positions[2][$this->position][0]) {
+                // whitespace_trim detected ({%-, {{- or {#-)
+                $text = rtrim($text);
+            } else {
+                // whitespace_line_trim detected ({%~, {{~ or {#~)
+                // don't trim \r and \n
+                $text = rtrim($text, " \t\0\x0B");
```

Comment:
I think this should consume one `\r\n` if they are immediately behind the opening brackets. That allows 
```
{{~
```
not to be an NOOP and do something that can be useful.
E.g.
```
abcd

{{~
```
Equivalent to:
```
abcd
{{
```

Reply:
I don't think so. `~` does/should not consume newlines to make the modifier super clear.

Reply:
I understand both parts of the argument and, tbh, I don't know which one is better.

1. Make it always functional by making it consume 1 "character" and then only line whitespace.
2. Make it do only 1 very specific thing.